### PR TITLE
Separate open and closed issue in metrics

### PR DESCRIPTION
**Pull request by romdum**
Hi,

My team and I need to separate closed and open issue in exposed metrics. 

The purpose is to have the number of open and closed issues by week in a Grafana dashboard.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Comment by lafriks** [comment]
I think old metric should be kept to not break compatibility

**Comment by zeripath** [comment]
Oh I missed that they'd removed that! Sorry for the bad review

File: models/models.go [hunk 1]
```diff
@@ -272,7 +272,7 @@ type Statistic struct {
 	Counter struct {
 		User, Org, PublicKey,
 		Repo, Watch, Star, Action, Access,
-		Issue, Comment, Oauth, Follow,
+		IssueClosed, IssueOpen, Comment, Oauth, Follow,
```
Comment (zeripath) [inline models/models.go]:
```suggestion
		Issue, IssueClosed, IssueOpen, Comment, Oauth, Follow,
```

File: modules/metrics/collector.go [hunk 2]
```diff
@@ -21,7 +21,8 @@ type Collector struct {
 	Comments      *prometheus.Desc
 	Follows       *prometheus.Desc
 	HookTasks     *prometheus.Desc
-	Issues        *prometheus.Desc
+	IssuesOpen    *prometheus.Desc
```
Comment (zeripath) [inline modules/metrics/collector.go:25]:
```suggestion
	Issues        *prometheus.Desc
	IssuesOpen    *prometheus.Desc
```

File: modules/metrics/collector.go [hunk 3]
```diff
@@ -72,9 +73,14 @@ func NewCollector() Collector {
 			"Number of HookTasks",
 			nil, nil,
 		),
-		Issues: prometheus.NewDesc(
-			namespace+"issues",
-			"Number of Issues",
+		IssuesOpen: prometheus.NewDesc(
```
Comment (zeripath) [inline modules/metrics/collector.go:82]:
```suggestion
		Issues: prometheus.NewDesc(
			namespace+"issues",
			"Number of Issues",
			nil, nil,
		),
		IssuesOpen: prometheus.NewDesc(
```

File: modules/metrics/collector.go [hunk 4]
```diff
@@ -164,7 +170,8 @@ func (c Collector) Describe(ch chan<- *prometheus.Desc) {
 	ch <- c.Comments
 	ch <- c.Follows
 	ch <- c.HookTasks
-	ch <- c.Issues
+	ch <- c.IssuesOpen
```
Comment (zeripath) [inline modules/metrics/collector.go:180]:
```suggestion
	ch <- c.Issues
	ch <- c.IssuesOpen
```

File: modules/metrics/collector.go [hunk 5]
```diff
@@ -217,9 +224,14 @@ func (c Collector) Collect(ch chan<- prometheus.Metric) {
 		float64(stats.Counter.HookTask),
 	)
 	ch <- prometheus.MustNewConstMetric(
-		c.Issues,
+		c.IssuesClosed,
```
Comment (zeripath) [inline modules/metrics/collector.go:239]:
```suggestion
		c.Issues,
		prometheus.GaugeValue,
		float64(stats.Counter.Issue),
	)
	ch <- prometheus.MustNewConstMetric(
		c.IssuesClosed,
```

**Comment by romdum** [comment]
If we keep Issue metrics, may be it's not necessary to keep IssueOpen metrics because: Issue - IssueClosed = IssueOpen ?

**Comment by zeripath** [comment]
So as these metrics are not produced within a transaction you can't actually assume that.

**Comment by zeripath** [comment]
OK I've made a little change which means that the counts are collected in one query.

**Review by lafriks — APPROVED** [review]

**Pull request merged**